pkg/rehearse/jobs.go

```diff
@@ -168,22 +180,28 @@ func filterPeriodics(changedPeriodics []prowconfig.Periodic, allowVolumes bool,
 	return periodics
 }
 
+func validateLabels(labels map[string]string) error {
+	if value, ok := labels[CanBeRehearsedLabel]; !ok || value != "true" {
+		return fmt.Errorf("job is not allowed to be rehearsed. Label %s is required", CanBeRehearsedLabel)
+	}
+	return nil
+}
+
 func filterJobSpec(spec *v1.PodSpec, allowVolumes bool) error {
 	// there will always be exactly one container.
 	container := spec.Containers[0]
 
-	if len(container.Command) != 1 || container.Command[0] != "ci-operator" {
-		return fmt.Errorf("cannot rehearse jobs that have Command different from simple 'ci-operator'")
-	}
-
-	for _, arg := range container.Args {
-		if strings.HasPrefix(arg, "--git-ref") || strings.HasPrefix(arg, "-git-ref") {
-			return fmt.Errorf("cannot rehearse jobs that call ci-operator with '--git-ref' arg")
-		}
-		if arg == "--promote" {
-			return fmt.Errorf("cannot rehearse jobs that call ci-operator with '--promote' arg")
+	if len(container.Command) > 0 && container.Command[0] == "ci-operator" {
+		for _, arg := range container.Args {
+			if strings.HasPrefix(arg, "--git-ref") || strings.HasPrefix(arg, "-git-ref") {
+				return fmt.Errorf("cannot rehearse jobs that call ci-operator with '--git-ref' arg")
+			}
+			if arg == "--promote" {
+				return fmt.Errorf("cannot rehearse jobs that call ci-operator with '--promote' arg")
+			}
```

Comment:
Personally I would drop these checks too and let the label have full control. Are we afraid of malice or mistakes? If malice, then it's still going to be possible to craft a PR that will run ci-op with promote. If mistakes, then I think it would be a unique mistake - generated jobs won't have the label so copy-paste should not be an issue, and nobody should ever write a `ci-operator --promote` job by hand...

Reply:
addresed

Reply:
I removed also the `git-ref` check, after the new changes in `ci-operator`